californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -94,9 +94,9 @@
 	private final AtomicReference<EndpointContext> destinationContext = new AtomicReference<EndpointContext>();
 
 	/** The type used for requests (CON is default) */
-	private Type type = Type.CON;
+	protected Type type = Type.CON;
```

Comment:
Why should this be more exposed than with useNONs/useCONs?

Reply:
This might be able to stay private. The tests runs fine with a small change on my end.

Reply:
OSCoreClient

```
    @Override
    protected Request send(Request request, Endpoint outEndpoint) {
        OSCoreEndpoint ep = (OSCoreEndpoint) outEndpoint;

        request.getOptions().addOption(new Option(OptionNumberRegistry.OSCORE));
        // use the specified message type
        request.setType(super.type);

        if (this.blockwise!=0) {
            request.getOptions().setBlock2(new BlockOption(BlockOption.size2Szx(this.blockwise), false, 0));
        }

        ep.sendRequest(request);
        return request;
    }
```

Any reason, why OSCORE requires to add the (default) type here? The original coap client just apply that type in some cases, but offer also advanced methods to send requests with an other type.

Reply:
Not anymore, i think. I will do some testing, but I think that I can remove it safely.